plugins/emoji/plugin.js

```diff
@@ -79,26 +152,466 @@
+						blockObject._.markItem( inputIndex );
+					} );
+
+					// Remove statusbar icons:
+					clearStatusbar();
+				};
+			}
+
+			function refreshNavigationStatus() {
+				var sections = blockElement.find( 'section' ).toArray(),
+					containerOffset = blockElement.findOne( '.cke_emoji-outer_emoji_block' ).getClientRect().top,
+					section,
+					groupName,
+					navigationElements;
+
+				section = CKEDITOR.tools.array.filter( sections, function( element ) {
+					var rect = element.getClientRect();
+					if ( !rect.height || element.findOne( 'h2' ).hasClass( 'hidden' ) ) {
+						return false;
+					}
+					return rect.height + rect.top > containerOffset;
+				} );
+				groupName = section.length ? section[ 0 ].data( 'cke-emoji-group' ) : false;
+				navigationElements = blockElement.find( 'nav li' ).toArray();
+
+				CKEDITOR.tools.array.forEach( navigationElements, function( node ) {
+					if ( node.data( 'cke-emoji-group' ) === groupName ) {
+						node.addClass( 'active' );
+					} else {
+						node.removeClass( 'active' );
+					}
+				} );
+			}
+
+			function updateStatusbar( element ) {
+				if ( element.getName() !== 'a' ) {
+					return;
+				}
+
+				blockElement.findOne( '.cke_emoji-status_icon' ).setText( escapeString( element.getText() ) );
+				blockElement.findOne( 'p.cke_emoji-status_description' ).setText( escapeString( element.data( 'cke-emoji-name' ) ) );
+				blockElement.findOne( 'p.cke_emoji-status_full_name' ).setText( escapeString( element.data( 'cke-emoji-full-name' ) ) );
+			}
+
+			function clearStatusbar() {
+				blockElement.findOne( '.cke_emoji-status_icon' ).setText( '' );
+				blockElement.findOne( 'p.cke_emoji-status_description' ).setText( '' );
+				blockElement.findOne( 'p.cke_emoji-status_full_name' ).setText( '' );
+			}
+
+			function clearSearchAndMoveFocus( event ) {
+				clearSearchInput();
+				moveFocus( event );
+			}
+
+			function moveFocus( event ) {
+				var groupName = event.data.getTarget().getAscendant( 'li', true ).data( 'cke-emoji-group' ),
+					firstSectionItem = blockElement.findOne( 'a[data-cke-emoji-group="' + escapeString( groupName ) + '"]' ),
+					itemIndex;
+
+				if ( !firstSectionItem ) {
+					return;
+				}
+
+				itemIndex = getItemIndex( blockObject._.getItems().toArray(), firstSectionItem );
+				firstSectionItem.focus( true );
+				blockObject._.markItem( itemIndex );
+			}
+
+			function getItemIndex( list, item ) {
+				return CKEDITOR.tools.array.indexOf( list, function( element ) {
+					return element.equals( item );
+				} );
+			}
+
+			function registerListeners( list ) {
+				CKEDITOR.tools.array.forEach( list, function( item ) {
+					var root = blockElement,
+						selector = item.selector,
+						listener = item.listener,
+						event = item.event;
+
+					CKEDITOR.tools.array.forEach( root.find( selector ).toArray(), function( node ) {
+						node.on( event, listener );
+					} );
+				} );
+			}
+
+			function escapeString( str ) {
```

Comment:
There is already a function for HTML encoding. Look in `CKEDITOR.tools` to reuse it.